vql/sigma/evaluator/modifiers/modifiers.go

```diff
@@ -67,14 +69,27 @@ func getComparator(comparators map[string]Comparator, modifiers ...string) (Comp
 			}
 		}
 		for _, modifier := range valueModifiers {
-			expected, err = modifier.Modify(expected)
+			newExpectedMod := []any{}
+			for _, expected := range expectedArr {
```

Comment:
This looks wrong - expectedArr is empty to start with so this loop doesnt do anything

Reply:
On line 63, `expectedArr` is initialized as `[]any{expected}`. Subsequently, each round of modifiers produces one or more potential values, which are temporarily stored in `newExpectedMod`, which replaces `expectedArr` after the inner loop.